src/context.c

```diff
@@ -414,3 +440,69 @@ xtt_get_my_pseudonym_lrsw(xtt_daa_pseudonym_lrsw *pseudonym_out,
+                                           unsigned char *in_buffer,
+                                           uint16_t in_buffer_size,
+                                           unsigned char *out_buffer,
+                                           uint16_t out_buffer_size,
+                                           xtt_version version,
+                                           xtt_suite_spec suite_spec)
+{
+    if (ctx_out == NULL)
+        return XTT_RETURN_NULL_BUFFER;
+
+    if (MAX_HANDSHAKE_SERVER_MESSAGE_LENGTH > in_buffer_size || MAX_HANDSHAKE_CLIENT_MESSAGE_LENGTH > out_buffer_size)
+        return XTT_RETURN_CONTEXT_BUFFER_OVERFLOW;
+
+    if (XTT_VERSION_ONE != version)
+        return XTT_RETURN_UNKNOWN_VERSION;
+
+    ctx_out->state = XTT_CLIENT_HANDSHAKE_STATE_START;
+
+    ctx_out->base.version = version;
+    ctx_out->base.suite_spec = suite_spec;
+    ctx_out->base.suite_ops = xtt_suite_ops_get(suite_spec);
+    if (NULL == ctx_out->base.suite_ops)
+        return XTT_RETURN_UNKNOWN_SUITE_SPEC;
+
+    ctx_out->base.in_buffer_start = in_buffer;
+    ctx_out->base.in_message_start = ctx_out->base.in_buffer_start;
+    ctx_out->base.in_end = ctx_out->base.in_buffer_start;
+    ctx_out->base.out_buffer_start = out_buffer;
+    ctx_out->base.out_message_start = ctx_out->base.out_buffer_start;
+    ctx_out->base.out_end = ctx_out->base.out_buffer_start;
+
+    xtt_crypto_hmac_init(&ctx_out->base.hash_out, ctx_out->base.suite_ops->hmac);
+    xtt_crypto_hmac_init(&ctx_out->base.inner_hash, ctx_out->base.suite_ops->hmac);
+    xtt_crypto_hmac_init(&ctx_out->base.prf_key, ctx_out->base.suite_ops->hmac);
+    xtt_crypto_hmac_init(&ctx_out->base.handshake_secret, ctx_out->base.suite_ops->hmac);
+
+    xtt_crypto_aead_key_init(&ctx_out->base.rx_key, ctx_out->base.suite_ops->aead);
+    xtt_crypto_aead_key_init(&ctx_out->base.tx_key, ctx_out->base.suite_ops->aead);
+    xtt_crypto_aead_nonce_init(&ctx_out->base.rx_iv, ctx_out->base.suite_ops->aead);
+    xtt_crypto_aead_nonce_init(&ctx_out->base.tx_iv, ctx_out->base.suite_ops->aead);
+
+    ctx_out->base.longterm_key_length = sizeof(xtt_ecdsap256_pub_key);
+    ctx_out->base.longterm_key_signature_length = sizeof(xtt_ecdsap256_signature);
+
+    ctx_out->base.tx_sequence_num = 0;
+    ctx_out->base.rx_sequence_num = 0;
+
+    ctx_out->verify_server_signature = verify_server_signature_ecdsap256;
+
+    ctx_out->copy_longterm_key = copy_longterm_key_ecdsap256;
+
+    ctx_out->compare_longterm_keys = compare_longterm_keys_ecdsap256;
+    ctx_out->copy_in_my_pseudonym = copy_in_pseudonym_client_lrsw;
+
+    if (0 != ctx_out->base.suite_ops->kx->keypair(&ctx_out->base.kx_pubkey,
+                                                  &ctx_out->base.kx_seckey))
+        return XTT_RETURN_CRYPTO;
+
+    if (0 != xtt_crypto_create_ecdsap256_key_pair(&ctx_out->longterm_key.ecdsap256, &ctx_out->longterm_private_key.ecdsap256))
```

Comment:
Don't need to generate the key pair in common.

Reply:
Sorry, didn't check the diff closely enough

Fixed now.

I diff'd this function with the `master` version of the `xtt_initialize_client_handshake_context`, to make sure this common function is doing exactly what that old function did except:
- Don't set the `ctx_out->longterm_sign`
- Don't generate the ECDSA key